Commands/InteractionCommands/ClearInteractions.cs

```diff
@@ -0,0 +1,264 @@
+                return;
+            }
+
+            if (count == 0 && upTo == "")
+            {
+                await ctx.FollowUpAsync(new DiscordFollowupMessageBuilder().WithContent($"{Program.cfgjson.Emoji.Error} I need to know how many messages to delete! Please provide a value for `count` or `up_to`.").AsEphemeral(true));
+                return;
+            }
+
+            // If count is too low or too high, refuse the request
+
+            if (count < 0)
+            {
+                await ctx.FollowUpAsync(new DiscordFollowupMessageBuilder().WithContent($"{Program.cfgjson.Emoji.Error} I can't delete a negative number of messages! Try setting `count` to a positive number.").AsEphemeral(true));
+                return;
+            }
+
+            if (count >= 1000)
+            {
+                await ctx.FollowUpAsync(new DiscordFollowupMessageBuilder().WithContent($"{Program.cfgjson.Emoji.Error} Deleting that many messages poses a risk of something disastrous happening, so I'm refusing your request, sorry.").AsEphemeral(true));
+                return;
+            }
+
+            // Get messages to delete, whether that's messages up to a certain one or the last 'x' number of messages.
+
+            if (upTo != "" && count != 0)
+            {
+                await ctx.FollowUpAsync(new DiscordFollowupMessageBuilder().WithContent($"{Program.cfgjson.Emoji.Error} You can't provide both a count of messages and a message to delete up to! Please only provide one of the two arguments.").AsEphemeral(true));
+                return;
+            }
+
+            List<DiscordMessage> messagesToClear;
+            if (upTo == "")
+            {
+                var messages = await ctx.Channel.GetMessagesAsync((int)count);
+                messagesToClear = messages.ToList();
+            }
+            else
+            {
+                DiscordMessage message;
+                ulong messageId;
+                if (!upTo.Contains("discord.com"))
+                {
+                    try
+                    {
+                        messageId = Convert.ToUInt64(upTo);
+                    }
+                    catch
+                    {
+                        await ctx.FollowUpAsync(new DiscordFollowupMessageBuilder().WithContent($"{Program.cfgjson.Emoji.Error} That doesn't look like a valid message ID or link! Try again."));
+                        return;
+                    }
+                }
+                else
+                {
+                    if (Constants.RegexConstants.discord_link_rx.Match(upTo).Groups[2].Value == ctx.Channel.Id.ToString())
+                    {
+                        messageId = Convert.ToUInt64(Constants.RegexConstants.discord_link_rx.Match(upTo).Groups[3].Value);
+                    }
+                    else
```

Comment:
By using `ulong.TryParse` in the `if` you can condense this and also avoid an uncaught `FormatError` when the message ID fails to parse for whatever arcane reason. (This should probably never happen but if it does then now it's caught and handled appropriately)
```suggestion
                    if (
                        Constants.RegexConstants.discord_link_rx.Match(upTo).Groups[2].Value != ctx.Channel.Id.ToString()
                        || !ulong.TryParse(Constants.RegexConstants.discord_link_rx.Match(upTo).Groups[3].Value, out messageId)
                    )
                    {
```